main/2025/CG-02.md

```diff
@@ -130,12 +198,2205 @@ sessions](https://calendar.google.com/calendar/u/0?cid=Y18zYmNkNzdiOWZmZjM4YzA0Y
+
+Sébastien Doeraene (in chat): https://webassembly.github.io/js-promise-integration/js-api/#jspi
+
+LW: Are there any additional implementation-specific limitations that we should take into account. You mentioned integers that looked like pointers, is that about conservative stack scanning? Are there other things, maybe CPU-specific considerations that we should know about?
+
+KM: We have concerns with future plans. Unfortunately I am not at liberty to disclose those plans.
+Largely our concern is in the intermingling of pointers and user-controlled integers, and the security implications of that. We have future security plans I can't go into that make that harder.
+It would make side stacks very hard, but wouldn’t necessarily affect e.g. how promises work 
+
+LW: If one were to run go natively, would that same thing prohibit Go from running natively?
+
+KW: I can’t talk about that. And I don't know much about Go.
+
+Deepti: I would encourage folks to go take a look at the longer response Francis wrote (https://github.com/WebKit/standards-positions/issues/422#issuecomment-2651632857). 
+The fact that the asynchronous interface is useful in itself. Would be an alternative to JSPI that doesn’t look like JSPI itself?
+
+Deepti: Conrad, I’d like to see a time box for what this discussion looks like, so we know what to expect.
+
+CW: I think this is something we should discuss over a reasonably short timeframe, e.g. by the next stack switching meeting we should have an idea of what the timeline would be
+
+KW: I did discuss this with Francis several months ago and I’m not aware of any progress since then.
+
+CW: I don’t want to be too much of a stickler for procedure, but we do want to have these discussion in public
+ 
+KW: there wasn’t a discussion on JSPI since then, so there wasn’t really an opportunity.
+
+David Thompson (in chat):  That's a good point re: webgpu. Realtime graphics on wasm is something I want to see improve.
+
+Reid Rankin (in chat): (I'm missing why JSPI requires side stacks. I don't want to take up the group's time here, but if anyone would  be willing to explain this to me offline or in a directed message I'd appreciate it.)
+
+David Thompson (in chat): I guess because you have to suspend the wasm program and resume it later to provide the illusion that you are doing synchronous execution?
+
+CW: A three-way poll (informal): F in favor of JSPI moving forward as is,
+N = Neutral,
+A = Against, want to consider alternatives such as what KM is proposing
+
+F: 25 in room, 12 in chat
+N: 6 in room, 7 in chat
+A: 1 in room, 1 in chat.
+
+CW: For the record, both people voting against are Apple representatives.
+
+KW: I think we’ve covered our concerns above.
+
+#### Stack Switching
+
+Sam Lindley (SL) presenting (TODO: slides)
```

Comment:
@slindley, can you provide a link to your slides, uploading them in a separate PR if necessary?

Reply:
PR https://github.com/WebAssembly/meetings/pull/1786